theano/gpuarray/type.py

```diff
@@ -250,6 +257,12 @@ def filter(self, data, strict=False, allow_downcast=None):
             if b and shp[i] != 1:
                 raise TypeError("Non-unit value on shape on a broadcastable"
                                 " dimension.", shp, self.broadcastable)
+        if not isinstance(data, gpuarray.GpuArray):
+            if old_data and old_data.shape == data.shape:
+                old_data.write(data)
```

Comment:
What are the restriction about the contiguous of data and old_data? See the issue about this and the link it contain. I think we are now less restrictive then in the past. If so, we should update the doc you can find from the issue.